app/helpers/bulk_downloads_helper.rb

```diff
@@ -56,4 +56,82 @@ def format_bulk_download(bulk_download, with_pipeline_runs = false)
     end
     formatted_bulk_download
   end
+
+  # This method models HeatmapHelper.sample_taxons_dict when fetching metrics. It uses the necessary logic from that function.
+  def self.generate_combined_sample_taxon_results_csv(samples, background_id, metric)
```

Comment:
are there other parts of this that could still be extracted into more shared helpers with `HeatmapHelper.sample_taxons_dict`? Just b/c these code paths tend to get out of sync very quickly

Reply:
or maybe moving just the metric formatting block into another function?

Reply:
This function doesn't use that much from `sample_taxons_dict`, despite it being the inspiration for this function. I only use `taxon_counts_cleanup` and `only_species_level_counts`.

I did add some additional comments and move the metric calculation into a separate method.